HW05 Jerry Lee.Rmd

```diff
@@ -0,0 +1,116 @@
+---
+title: "HW05 Jerry Lee"
+author: "jerry lee"
+date: "8/7/2020"
+output: html_document
+---
+
+# HW05
+Due August 3rd
+
+### HW Assignment
+
+This week the goal is to practice tidying data and joining files together. Similar to last week you can practice these skills with data you already have or you can follow with the two (probably pretty short) HW assignments below.
+
+## 1. Tidying the dad_mom file
+
+In this repo, you'll find a dad_mom.txt file that isn't particularly tidy. Clean this up so there are only 4 columns 
+Tidy this data frame so that it adheres to the tidy data principles:
+
+    Each variable must have its own column.
+    Each observation must have its own row.
+    Each value must have its own cell.
+
+```{r loading packages}
+library(tidyverse)
+library(ggplot2)
+library(forcats)
+```
+
+```{r reading data #1}
+dad_mom<-read_tsv("dad_mom.txt")
+
+dad_mom
+```
+
+```{r tidying data}
+dad_mom_tidy<-dad_mom%>%
+  unite("dad",name_dad:income_dad,sep="_")%>% #lines 43-44 to associate incomes with each person
+  unite("mom",name_mom:income_mom,sep="_")%>%
+  gather(key="parent_type",value="name",dad,mom)%>% #turn each row into an observation
+  separate(name,c("name","income"),sep="_") #create separate columns for name and income variables
+
+dad_mom_tidy
+```
+
+## 2. Joining together CTRP data
+
+CTRP is a dataset I use in my research. CTRP screens cancer cells with various compounds and determines the Area Under the Dose Response Curve (AUC, a metric for how sensitive a cell line is to that compound). However, when you download the data, it comes in parts that requires a lot of id matching to be able to use reasonably easily. For example, the file that contains the AUC data doesn't tell you the cell line name or the type of cancer it is, it just gives an experiment ID number. If you wanted to know which cell line was which, you'd have to reference the experiment_info.csv file which gives you a cell line ID for each experiment, and then reference the the cancer_cell_line_file.csv to figure out what the name of that cell line actually is. 
+
+That is all to say, it would be much easier if those files were all together instead. You're goal is to join together the 5 csv together.
+
+```{r reading data #2}
+auc1<-read_csv("CTRP_files/AUC_1.csv")
+auc2<-read_csv("CTRP_files/AUC_2.csv")
+ccl<-read_csv("CTRP_files/cancer_cell_line_info.csv")
+cpd<-read_csv("CTRP_files/compound_info.csv")
+expt<-read_csv("CTRP_files/Experiment_info.csv")
+```
+```{r joining data}
+joined<-expt%>%
+  full_join(ccl,by="master_ccl_id")%>%
```

Comment:
I'm just wondering why you chose to use full_join() instead of inner_join? I don't remember the difference. Your method definitely appears to use more efficient code.